indico/modules/users/blueprint.py

```diff
@@ -58,6 +57,8 @@
     _bp.add_url_rule('/emails/<email>', 'user_emails_delete', RHUserEmailsDelete, methods=('DELETE',))
     _bp.add_url_rule('/emails/make-primary', 'user_emails_set_primary', RHUserEmailsSetPrimary, methods=('POST',))
 
+_bp.add_url_rule('/<int:user_id>/dashboard.ics', 'export_dashboard_ics', RHExportDashboardICS)
```

Comment:
If we use an unique-enough name like `user_token` we could even have an `RHTokenProtected` base RH that takes care of assigning the user and checking.

On a side note, this is where it sucks a bit that we went for `session.user` as the way to get the current user. Had we used a `current_user` proxy (and template global) we could just populate that when a token has been used and thus have access to the current user in the exact same way regardless of how they are authenticated (token/cookie/apikey/whatever)

Reply:
What would be the advantage of going that way?

Reply:
- not having to show the userid in the URL (it kind of invites people to try to change it, even though it of course won't do anything nasty)
- URL looks simpler - but since some of them have a query string anyway I guess it doesn't matter that much
- not sure if there aren't some calendaring tools that don't like query strings

anyway, I guess it boils down to personal preference

---

but regardless of the URL style, I would move the logic that checks the token into a custom RH subclass

Reply:
I think that adding the token to the URL and removing the user makes the API less coherent. And if we ever implement the `current_user` proxy you talk about, then it shouldn't matter which auth method they're using, that should be the URL.